### PR TITLE
fix(objectschema)!: omit id from schema returned by without + only

### PR DESCRIPTION
**Pull request by esatterwhite**
the methods without and only return new schema objects with the same id that was found on the original. In terms of schema reuse, this creates two different schemas with the same id which is technically invalid. Additionally, it was not possible to set the id property on an object schema that had properties defined. This problem originates from the `setattribute` function which will always set the attribute on the schema properties if they are defined. In the case of the object schema, this is almost always the case. This changes the id function on the object schema to always generate a new schema with the id set on the object schema rather than its properties

BREAKING CHANGE: ObjectSchema.id() will always set the id on the root object
BREAKING CHANGE: ObjectSchema.without() will omit id from the return schema
BREAKING CHANGE: ObjectSchema.only() will omit id from the return schema

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by esatterwhite** [comment]
anything left to do here? or does anyone else need to look at this?

**Comment by esatterwhite** [comment]
@aboutlo ?

**Comment by esatterwhite** [comment]
@aboutlo thoughts?

**Comment by aboutlo** [comment]
Hey @esatterwhite thank you for your PR 💪 

Could you add the changes in the documentation aka:
- add the documentation tags and explain the different behaviours 
- update the doc by running `npm run doc` and commit the markdown updated

Then I can merge and make a release with these breaking changes and the drop of the old node versions

File: .nvmrc [hunk 1]
```diff
@@ -1 +1 @@
-v12.14.0
+v14.19.0
```
Comment (esatterwhite) [inline .nvmrc:1]: This doesn't actually work on node 12 anymore

**Pull request merged**